### DISCUSSION TITLE
Why is plot() not plotting certain layers from gf.grid()?

### DISCUSSION THREAD

**Discussion by ghost** [gdsfactory/gdsfactory#3237] [Q&A] [answered]
_edited in an attempt to follow Best Practices_

When doing a quick `.plot()` to check results, I see that certain layers are not being plotted, while `write_gds()` output is fine. I do not understand the reason.

Is this just me, or is this reproducible?

```
import gdsfactory as gf
@gf.cell
def swatch(beta, index):
    _ = gf.components.rectangle(size = (beta*3,beta*3), layer = (index+1,0))
    return _ 

def make_list(rows: int, columns: int,beta: float = 1) -> list:
    list = []
    for i in range(((rows*columns))):
        _ = swatch(beta=beta, index=i)
        list.append(_)
    return list

@gf.cell
def layered_rec_grid(rows: int, columns: int,beta: float = 1) -> gf.typings.Component:
    list = make_list(rows = rows, columns = columns, beta = beta)
    g = gf.grid(
        list,
        spacing=(3*beta, 3*beta),
        shape=(rows, columns),
        align_x="x",
        align_y="y",
    )  
    return g

@gf.cell
def reference_dose_bars(rows: int, columns: int,beta: float = 1) -> gf.typings.Component:
    return gf.components.rectangle(size=(3*beta,(beta*rows*7)), layer=(((rows*columns)+1), 0))

@gf.cell
def dose_grid(rows: int, columns: int,beta: float = 1) -> gf.typings.Component:
    top = gf.Component()
    ref1 = top << layered_rec_grid(rows = rows,columns = columns, beta=beta) 
    bar = reference_dose_bars(rows = rows,columns = columns, beta=beta)
    ref1 = top << bar
    ref1.dmove([-6*beta, 0])
    ref2 = top << bar
    ref2.dmove([6*beta*columns, 0])
    return top


beta = 13
final = dose_grid(rows=3,columns=10,beta=beta)
final.plot()
```

**Comment by luomeng007** [comment]
Hallo, bro, I also find this problem the other day. I think this is due to the default setting about GDSFactory. The GDSFactory has its own predefined layers. You can check them from this link [generic layers of GDSFactory](https://gdsfactory.github.io/gdsfactory/notebooks/03_layer_stack.html). Only these layers can be shown by using plot() method. For structure on the other layers, you do not need to export them to .gds file to check them. Just open your KLayout software and keep it open. Then use show() method instead of plot() method. After running you code, you can get the structure in KLayout view.

**Comment by joamatab** [comment]
Thanks for pointing this out

I can send a fix for this and release a new gdsfactory version

**Reply by joamatab** [reply, marked as answer]
See fix here
https://github.com/gdsfactory/gdsfactory/pull/3249

I can release a new version of gdsfactory that you can get when you do `pip install gdsfactory --upgrade`
